### DISCUSSION TITLE
Is there a way to support full fuzzy search

### DISCUSSION THREAD

**Discussion by yugj** [juicedata/juicefs#5035] [General]
**What would you like to be added**:
I want to fuzzy query file information, not just through left matching
**Why is this needed**:

I know that s3 protocol only supports left match fuzzy search，however, juicefs metadata stores full file information and theoretically supports full fuzziness.

**Comment by davies** [comment]
Can you give some examples and also the potential use case?

**Reply by yugj** [reply]
for example, juicefs stores a batch of files, such as: `s3gateway.txt, s4gateway.txt, s5gateway.txt`, etc.
user searches expect to find the file through the `gateway` string, but now s3 gateway only supports left matching query, making it difficult to obtain accurate results.
